.github/ServiceX/python/transformer.py

```diff
@@ -0,0 +1,187 @@
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import json
+import time
+
+from servicex.transformer.servicex_adapter import ServiceXAdapter
+from servicex.transformer.transformer_argument_parser import TransformerArgumentParser
+from servicex.transformer.kafka_messaging import KafkaMessaging
+from servicex.transformer.object_store_manager import ObjectStoreManager
+from servicex.transformer.rabbit_mq_manager import RabbitMQManager
+from servicex.transformer.uproot_events import UprootEvents
+from servicex.transformer.uproot_transformer import UprootTransformer
+from servicex.transformer.arrow_writer import ArrowWriter
+import uproot
+import os
+import sys
+import traceback
+import pyarrow as pa
+
+
+# How many bytes does an average awkward array cell take up. This is just
+# a rule of thumb to calculate chunksize
+avg_cell_size = 42
+MAX_RETRIES = 3
+
+messaging = None
+object_store = None
+
+
+# noinspection PyUnusedLocal
+def callback(channel, method, properties, body):
+    transform_request = json.loads(body)
+    _request_id = transform_request['request-id']
+    _file_path = transform_request['file-path'].encode('ascii', 'ignore')
+    _file_id = transform_request['file-id']
+    _server_endpoint = transform_request['service-endpoint']
+    # _chunks = transform_request['chunks']
+    servicex = ServiceXAdapter(_server_endpoint)
+
+    servicex.post_status_update(file_id=_file_id,
+                                status_code="start",
+                                info="xAOD Transformer")
+
+    tick = time.time()
+    file_done = False
+    file_retries = 0
+    while not file_done:
+        try:
+            # Do the transform
+            root_file = _file_path.replace('/', ':')
+            output_path = '/home/atlas/' + root_file
```

Comment:
do we really have to use `/home/atlas/`?

Reply:
Will change this to `/home/cmsuser/`